lighthouse-core/audits/seo/font-size.js

```diff
@@ -121,80 +120,84 @@ function nodeToTableNode(node) {
-      source: 'User Agent Stylesheet',
     };
   }
 
-  if (styleDeclaration.type === 'Regular' && styleDeclaration.parentRule) {
+  // Combine all the selectors for the associated style rule
+  // example: .some-selector, .other-selector {...} => `.some-selector, .other-selector`
+  let selector = '';
+  if (styleDeclaration.parentRule) {
     const rule = styleDeclaration.parentRule;
-    const stylesheet = styleDeclaration.stylesheet;
-
-    if (stylesheet) {
-      let source;
-      const selector = rule.selectors.map(item => item.text).join(', ');
-
-      if (stylesheet.sourceURL) {
-        const url = new URL(stylesheet.sourceURL, baseURL);
-        const range = styleDeclaration.range;
-        source = `${url.href}`;
-
-        // !!range == has defined location in a source file (.css or .html)
-        if (range) {
-          let line = range.startLine + 1;
-          let column = range.startColumn;
-
-          // Add the startLine/startColumn of the <style> element to the range, if stylesheet
-          // is inline.
-          // Always use the rule's location if a sourceURL magic comment is
-          // present (`hasSourceURL` is true) - this makes the line/col relative to the start
-          // of the style tag, which makes them relevant when the "file" is open in DevTool's
-          // Sources panel.
-          const addHtmlLocationOffset = stylesheet.isInline && !stylesheet.hasSourceURL;
-          if (addHtmlLocationOffset) {
-            line += stylesheet.startLine;
-            // The column the stylesheet begins on is only relevant if the rule is declared on the same line.
-            if (range.startLine === 0) {
-              column += stylesheet.startColumn;
-            }
-          }
-
-          source += `:${line}:${column}`;
-        }
-      } else {
-        // dynamically injected to page
-        source = 'dynamic';
-      }
+    selector = rule.selectors.map(item => item.text).join(', ');
+  }
 
-      return {
-        selector,
-        source,
-      };
+  if (styleDeclaration.stylesheet && !styleDeclaration.stylesheet.sourceURL) {
+    // Dynamically injected into page.
+    return {
+      source: {type: 'code', value: 'dynamic'},
+      selector,
+    };
+  }
+
+  // !!range == has defined location in a source file (.css or .html)
+  // sourceURL == stylesheet URL || raw value of magic `sourceURL` comment
+  // hasSourceURL == flag that signals sourceURL is the raw value of a magic `sourceURL` comment, *not* a real resource
+  if (styleDeclaration.stylesheet && styleDeclaration.range) {
+    const {range, stylesheet} = styleDeclaration;
+
+    // DevTools protocol does not provide the resource URL if there is a magic `sourceURL` comment.
+    // `sourceURL` will be the raw value of the magic `sourceURL` comment, which likely refers to
+    // a file at build time, not one that is served over the network that we could link to.
+    const urlIsNetworkResource = !stylesheet.hasSourceURL;
```

Comment:
i'm not sure this equivalence is true.  

1) an stylesheet stylesheet can be served over the network but then provide its own sourceURL.
2) a sourceURL can point to `doesntexist.css` or it can point to `/css/style.css` which may be a real network-accessible file.

Reply:
> an stylesheet stylesheet can be served over the network but then provide its own sourceURL.

Yes - and b/c of a failing in the protocol we can't figure out the true network url directly from the `CSSStyleSheetHeader`. I suppose we could look at the `ownerNode` and get the URL used in the `link` element.

But then we'd need to double the urls collected here, since we'd need one that is a "network url" (to provide a click-thru link) and the other that is a devtools ui location url (for opening in source panel - must use the sourceURL if it exists).

> a sourceURL can point to doesntexist.css __or it can point to /css/style.css which may be a real network-accessible file__.

I think this must be very rare in practice. Why would someone ensure the sourceURL references a real network resource? It just changes the name as seen in DevTools, so I don't see why someone would also make sure it is a real network resource. If it were, why wouldn't they just reference it directly?

___

The goal here is to avoid rendering a link for a URL that is not web-accessible. The only signal I see for that is `hasSourceURL`, and the fact that using it to reference a real network resource is an unexpected (and I think pointless) use case.

Although I see that using the name `urlIsNetworkResource` is a stronger assertion than it needs to be (after all, it possible could be a real url). Maybe just a rename? `urlIsFromSourceUrlComment`, `urlIsFromComment`?